core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,959 @@
+                // 25th percentile
+                {
+                    data: function (row, type) {
+                        return displayRowsForSummaryMetricsTable(row, type, 1);
+                    }
+                },
+                // Median
+                {
+                    data: function (row, type) {
+                        return displayRowsForSummaryMetricsTable(row, type, 2);
+                    }
+                },
+                // 75th percentile
+                {
+                    data: function (row, type) {
+                        return displayRowsForSummaryMetricsTable(row, type, 3);
+                    }
+                },
+                // Max
+                {
+                    data: function (row, type) {
+                        return displayRowsForSummaryMetricsTable(row, type, 4);
+                    }
+                }
+            ],
+            "columnDefs": [
+                { "type": "duration", "targets": 1 },
+                { "type": "duration", "targets": 2 },
+                { "type": "duration", "targets": 3 },
+                { "type": "duration", "targets": 4 },
+                { "type": "duration", "targets": 5 }
+            ],
+            "paging": false,
+            "searching": false,
+            "order": [[0, "asc"]],
+            "bSort": false
+        };
+        taskSummaryMetricsDataTable = $(taskMetricsTable).DataTable(task_conf);
+    }
+    task_summary_metrics_table_current_state_array = task_summary_metrics_table.slice();
+}
+
+function createRowMetadataForColumn(columnName, data, checkboxId) {
+  var row = {
+      "metric": columnName,
+      "data": data,
+      "checkboxId": checkboxId
+  };
+  return row;
+}
+
+function reselectCheckboxesBasedOnTaskTableState() {
+    var allChecked = true;
+    var task_summary_metrics_table_current_filtered_array = task_summary_metrics_table_current_state_array.slice();
+    if (typeof taskTableSelector !== 'undefined' && task_summary_metrics_table_current_state_array.length > 0) {
+        for (k = 0; k < optionalColumns.length; k++) {
+            if (taskTableSelector.column(optionalColumns[k]).visible()) {
+                $("#box-"+optionalColumns[k]).prop('checked', true);
+                task_summary_metrics_table_current_state_array.push(task_summary_metrics_table_array.filter(row => (row.checkboxId).toString() == optionalColumns[k])[0]);
+                task_summary_metrics_table_current_filtered_array = task_summary_metrics_table_current_state_array.slice();
+            } else {
+                allChecked = false;
+            }
+        }
+        if (allChecked) {
+            $("#box-0").prop('checked', true);
+        }
+        createDataTableForTaskSummaryMetricsTable(task_summary_metrics_table_current_filtered_array);
+    }
+}
+
+function getStageAttemptId() {
+  var words = document.baseURI.split('?');
+  var attemptIdStr = words[1].split('&')[1];
+  var digitsRegex = /[0-9]+/;
```

Comment:
perhaps we should actually just grab the section between & and #? 

we might put in a comment about why we do this since urls can have format: /proxy/application_1539986433979_27115/stages/stage/?id=0&attempt=0#tasksTitle

Reply:
I used regex so that if there might be a different character in the url instead of #, like another & added someday, the code should work regardless. Let me know if that is an issue. Have added the comment.